idpartner/__init__.py

```diff
@@ -0,0 +1,234 @@
+            })
+            public_jwks.append(public_jwk)
+
+        return {"keys": public_jwks}
+
+    def get_authorization_url(
+        self, query, proofs, scope, extra_authorization_params={}
+    ):
+        if "iss" not in query:
+            return f"{self.config.get('account_selector_service_url')}/auth/select-accounts?client_id={self.config.get('client_id')}&visitor_id={query.get('visitor_id')}&scope={scope}&claims={'+'.join(self._extract_claims(extra_authorization_params.get('claims', [])))}"
+
+        self.config["iss"] = query.get("iss")
+        self._init_client(query.get("iss"))
+
+        extra_authorization_params["claims"] = json.dumps(
+            extra_authorization_params.get("claims")
+        )
+        extended_authorization_params = {
+            "redirect_uri": self.config.get("redirect_uri"),
+            "code_challenge_method": "S256",
+            "code_challenge": create_s256_code_challenge(proofs.get("code_verifier")),
+            "state": proofs.get("state"),
+            "nonce": proofs.get("nonce"),
+            "scope": scope,
+            "response_type": "code",
+            "client_id": self.config.get("client_id"),
+            "x-fapi-interaction-id": str(uuid.uuid4()),
+            "identity_provider_id": query.get("idp_id"),
+            "idpartner_token": query.get("idpartner_token"),
+            "response_mode": "jwt",
+        }
+
+        pushed_authorization_request_params = extended_authorization_params
+        if self.config.get("jwks"):
+            pushed_authorization_request_params = {
+                "request": self._create_request_object(extended_authorization_params)
+            }
+
+        request_uri = self._push_authorization_request(
+            pushed_authorization_request_params
+        ).get("request_uri")
+        return (
+            self.oauth_client.server_metadata.get("authorization_endpoint")
+            + f"?request_uri={request_uri}"
+        )
+
+    def token(self, query, proofs):
+        jwt = query.get("response")
+
+        code = self._decode_jwt(jwt).get("code")
+        return self.oauth_client.fetch_access_token(
+            code=code,
+            code_verifier=proofs.get("code_verifier"),
+        )
+
+    def userinfo(self, token):
+        return self.oauth_client.userinfo(token=token)
+
+    def _decode_jwt(self, jwt):
+        if len(jwt.split("."))==5:
+            jwe = jose.JsonWebEncryption()
+            enc_key = self._get_key_by_use("enc")
+            result = jwe.deserialize_compact(jwt, enc_key)
+            jwt = result.get("payload").decode("utf-8")
+        jwks = self.oauth_client.fetch_jwk_set()
+        return jose.jwt.decode(jwt, jwks)
+
+    def _create_request_object(self, params):
+        extended_params = {
+            **params,
+            **{
+                "iss": self.config["client_id"],
+                "aud": self.config["iss"],
+                "exp": int((datetime.now() + timedelta(minutes=1)).timestamp()),
```

Comment:
🤔  can you please double check this is the format we need (epoch)? I tried it and I get something that seems to be incorrect `print((datetime.now() + timedelta(minutes=1)).timestamp()) =>     
1701135065.088601`

Reply:
Same for other occurrences of `datetime.now().timestamp()`

Reply:
you were omitting the cast to int

Reply:
i compared with ruby one